docs/Project-Configuration/Github-Integrations.md

```diff
@@ -174,6 +173,66 @@ The check run will include a `view more details` link which will link back to th
+
+If you don't already have a github app, reach out to the IT team for help obtaining one.
+
+#### Saving An App Id and Key
+
+To get started, save your app ID and app key in your project settings in the GitHub App Settings tab. ![save-app.png](../images/github_app_id_and_key.png)
+
+Once an app id and key are saved, the app id will remain visible, but the app key will be redacted. If you need to make any changes to it, you can delete it and save it again.
+
+#### Restricting Generated Tokens
+
+When tokens are generated by the github.generate_token project command, they will have the full permissions of the github app.
+
+> While an owner and repository is required when generating a token from a github app (the project command will default to the owner and repository of the project if not specified), you cannot rely on the token being scoped down to that repository, as it may have the power to access other repositories in the org as well.
+
+Tokens generated by the github.generate_token can optionally be restricted directly in the github.generate_token command (link coming soon) and in project settings.
+
+To restrict it in project settings, start by creating a GitHub permission group in the GitHub Permission Groups tab. For a list of available permission types and levels, please take a look at `properties of permissions` in [the github documentation](https://docs.github.com/en/rest/apps/apps?apiVersion=2022-11-28#create-an-installation-access-token-for-an-app).
+Creating a permission group will on it's own not have any affect until it's applied to a requester type.
+
+![permission-group.png](../images/github_permission_groups.png)
+
+Once you have created a permission group, head back over to the GitHub App Settings page and assign the group to all the requester types that you want to restrict to that permission group. For example, if you select "clone only" from the dropdown for `GitHub Pull Request`, any tokens generated by pull request patches will only have `contents:read` permissions and nothing else.
+
+![github token requester type](../images/github_token_requester.png)
+
+## Tokens Restricted in Both Project Settings and The Command
+
+If token restrictions are specified both in the project configuration file under the `github.generate_token` project command directly and in project settings, the generated token will have the most restrictive intersection of the two.
+
+For example, if the project settings for the requester specifies
+
+```
+{
+    secrets: admin
+    pull-request: write
+    actions: read
+}
+```
+
+and the command specifies
+
 ```
+{
+    pull-request: read
+    actions: write
+}
+```
+
+The permissions for the resulting token will be
 
 ```
+# note that secrets not being specified is the
```

Comment:
Should this comment specifically say secrets? I think it sounds confusing since the word secret is usually 'special' and it might be lost on users that really it means any field, maybe if we add another one like "checks: write" in the 2nd one and say secrets and checks in this comment?

Reply:
Good point, thank you! I swapped out secrets for checks